src/libs/ActiveClientManager/index.ts

```diff
@@ -45,13 +45,15 @@ Onyx.connect({
     },
 });
 
-let wasTheLeader = false;
+let isPromotingNewLeader = false;
 
 /**
  * The last GUID is the most recent GUID, so that should be the leader
  */
 const isClientTheLeader: IsClientTheLeader = () => {
-    if (wasTheLeader) {
+    // keep reporting currently leading client on page refresh
+    // when clientID is already removed from activeClients list
```

Comment:
```suggestion
    // When a new leader is being promoted, there is a small period of time when the current leader's clientID has been removed from the activeClients list due to asynchronous operations, but the new leader hasn't officially taken over yet. This can lead to a moment when the page is refreshed where multiple leaders are being reported. This early return here will prevent that from happening by keeping the current leader as the "active leader" until the other leader is fully promoted.
```
Please be sure that comments are all proper English and do a thorough job of explaining the purpose of the logic. Also, even though I've added this suggestion, you'll need to split it up into multiple lines (I suggest using our line max length setting for this).

Reply:
Thank you, I've updated the comment.